src/ape/utils/basemodel.py

```diff
@@ -74,7 +76,36 @@ def provider(self) -> "ProviderAPI":
             raise ProviderNotConnectedError()
         return self.network_manager.active_provider
 
-    def create_contract_container(self, contract_type: "ContractType") -> "ContractContainer":
+    def create_contract_type(self, address: AddressType) -> Optional[ContractType]:
+        """
+        Helper method for creating a ``ContractType`` from an address.
+
+        Args:
+            address (AddressType): The address of the deployed contract.
+
+        Returns:
+            ContractType
+        """
+        network = self.provider.network
+        address = self.conversion_manager.convert(address, AddressType)
+        contract_type = None
+
+        if address in network.contract_cache:
+            # Check contract cache (e.g. previously deployed/downloaded contracts)
+            contract_type = network.contract_cache[address]
```

Comment:
The idea was eventually to store these on disk, but this works for now

Directory structure:

```
.ape/
  [ecosystem-name]/
    [network-name]/
      contract-types/
        [address].json  # ABI-json file
```

Reply:
This directory structure seems opposite? We need to be able to look up a contract type by address but this structure is for looking up the address of a contract type.

We get an address from the trace and need to be able to look it up. Also, caching the results from explorer.get_contract_type(address).

Reply:
```py
json_abi = json.loads(Path(f"~/.ape/{ecosystem_name}/{network_name}/contract-types/{address}.json").read_text())
```